responses/__init__.py

```diff
@@ -341,6 +344,25 @@ def _req_attr_matches(match, request):
 
         return True, ""
 
+    def _req_kwargs_matches(self, kwargs):
+        """
+        Check that arguments provided to requests method match response
+        :param kwargs: (dict) values provided to requests method
+        :return:
+            (bool) True if all args match
+            (str) reason with info which args failed
+        """
+        for arg in kwargs:
+            if hasattr(self, arg):
+                response_arg_val = getattr(self, arg)
+                requests_arg_val = kwargs[arg]
+                if response_arg_val != requests_arg_val:
```

Comment:
This seems like it could cause existing mocks to start failing if the requests for those mocks don't have all the same arguments. Should we provide this as an opt in matcher like `responses.matchers.all_kwargs_matcher()` instead?

Reply:
idea here is to really match the request that was requested. Otherwise, you may expect that you request `stream=True` but in fact you get no stream.
in such a case we will raise, this will be applied only to attributes that are assigned on the Response object, otherwise they are passed through

I can change the behavior to use matchers, but then I would need to change signature of the `match()` function to take kwargs as well
OR
we attach kwargs to request object before it is gone for validation, I need to check `requests` module how they handle kwargs

so, which option will you prefer: 1, 2, or 3 ? :D

Reply:
Attaching the kwargs to the simulated request and using a matcher would be preferrable. We're already mutating the request to add `request.params` and the `match` API allows us to keep the logic for kwarg matching in one place, and makes enabling this feature explicit and straightforward to the reader of test code.

Reply:
done